src/file_cache.rs

```diff
@@ -96,21 +96,25 @@ impl<T: AsyncFromStrWithState> FileCache<T> {
     }
 
     pub async fn get(&self, app_state: &AppState, path: &PathBuf) -> anyhow::Result<Arc<T>> {
-        log::trace!("Attempting to get from cache {:?}", path);
-        if let Some(cached) = self.cache.read().await.get(path) {
+        if let Some(cached) = self.cache.get(path) {
```

Comment:
I think we just need to clone the result of `self.cache.get` before if..let here.